lib/new_relic/transaction/complete.ex

```diff
@@ -170,6 +159,40 @@ defmodule NewRelic.Transaction.Complete do
     {[segment_tree], tx_attrs, tx_error, span_events, apdex, tx_metrics}
   end
 
+  defp transform_root_process_segment(tx_attrs) do
+    transformed =
+      tx_attrs
+      |> Map.take([:name, :pid, :start_time, :end_time])
+      |> transform_trace_time_attrs(tx_attrs.start_time)
+      |> transform_trace_name_attrs()
+
+    struct(Transaction.Trace.Segment, transformed)
+  end
+
+  defp transform_function_segments(function_segments, start_time) do
+    for segment <- function_segments do
+      segment = transform_function_segment(segment, start_time)
+      struct(Transaction.Trace.Segment, segment)
+    end
+  end
+
+  defp transform_function_segment(segment, start_time) do
+    segment
+    |> transform_time_attrs()
+    |> transform_trace_time_attrs(start_time)
+    |> transform_trace_name_attrs()
+  end
```

Comment:
No need for another intermediate function here...

```suggestion
  defp transform_function_segments(function_segments, start_time) do
    for segment <- function_segments do
      segment =
        segment
        |> transform_time_attrs()
        |> transform_trace_time_attrs(start_time)
        |> transform_trace_name_attrs()

      struct(Transaction.Trace.Segment, segment)
    end
  end
```